File: apps/desktop/patch_logo.js
const fs = require('fs');
const path = 'c:/Users/akula/Downloads/Hide-WIN/Hide-Win-Master/src/components/app/HideWinApp.js';
let content = fs.readFileSync(path, 'utf8');

const targetStr = `<!-- Posh Gear Logo (Like Session Window) -->
                        <div style="display: flex; align-items: center; justify-content: center; width: 28px; height: 28px; background: rgba(255,255,255,0.08); border: 1px solid rgba(255,255,255,0.05); border-radius: 50%; color: rgba(255,255,255,0.9); margin-left: 2px;">
                            <svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
                                <circle cx="12" cy="12" r="5"></circle>
                                <line x1="12" y1="1" x2="12" y2="3"></line>
                                <line x1="12" y1="21" x2="12" y2="23"></line>
                                <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
                                <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
                                <line x1="1" y1="12" x2="3" y2="12"></line>
                                <line x1="21" y1="12" x2="23" y2="12"></line>
                                <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
                                <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
                            </svg>
                        </div>`;

const replaceStr = `<!-- App Logo (White for dark bg) -->
                        <div style="width: 26px; height: 26px; background-image: url('./assets/images/logo_white.png'); background-size: contain; background-position: center; background-repeat: no-repeat; margin-left: 6px; margin-right: 2px;">
                        </div>`;

if (content.includes(targetStr)) {
    content = content.replace(targetStr, replaceStr);
    fs.writeFileSync(path, content, 'utf8');
    console.log("Successfully patched logo!");
} else {
    console.log("Could not find target string.");
}

File: patch_logo.py
import re
with open('Hide-Win-Master/src/components/app/HideWinApp.js', 'r', encoding='utf-8') as f:
    text = f.read()

old_logo = '''                    <div style="display:flex;align-items:center;min-width:0;flex:1;pointer-events:none;">
                        ${logoUrl ? html`<img src=${logoUrl} alt="HideWin" style="display:block;max-width:120px;width:auto;height:36px;object-fit:contain;object-position:left center;" />` : ''}
                    </div>'''

new_logo = '''                    <div style="display:flex;align-items:center;min-width:0;flex:1;pointer-events:none;">
                        ${logoUrl ? html`<img src=${logoUrl} alt="HideWin" style="display:block;max-width:120px;width:auto;height:36px;object-fit:contain;object-position:left center;" />` : html`<span style="font-size:17px;font-weight:700;letter-spacing:-0.3px;margin-left:4px;user-select:none;">HideWin</span>`}
                    </div>'''

text = text.replace(old_logo, new_logo)

with open('Hide-Win-Master/src/components/app/HideWinApp.js', 'w', encoding='utf-8') as f:
    f.write(text)
